_playwright-tests/UI/IntrospectRepo.spec.ts

```diff
@@ -0,0 +1,85 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { closePopupsIfExist, getRowByName } from './helpers/helpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+
+test.describe('Introspect Repositories', () => {
+  const repoName = 'introspection-test';
+  const repoUrl = 'https://fedorapeople.org/groups/katello/fakerepos/zoo/';
+
+  test.beforeEach(async ({ page }) => {
+    await test.step('Navigate to the repository page', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+    });
+  });
+
+  test('Create an introspection repository', async ({ page }) => {
+    await test.step('Cleanup repository, if using the same url', async () => {
+      await deleteAllRepos(page, `&url=${repoUrl}`);
+    });
+
+    await test.step('Open the add repository modal', async () => {
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByText('Add custom repositories')).toBeVisible();
+    });
+
+    await test.step('Fill the create repository form', async () => {
+      await page.getByPlaceholder('Enter name').fill(repoName);
+      await page.getByPlaceholder('https://').fill(repoUrl);
+    });
+
+    await test.step('Submit the form and wait for modal to disappear', async () => {
+      await Promise.all([
+        page.getByRole('button', { name: 'Save' }).click(),
+        page.waitForResponse(
+          (resp) =>
+            resp.url().includes('/bulk_create/') && resp.status() >= 200 && resp.status() < 300,
+        ),
+        expect(page.getByText('Add custom repositories')).not.toBeVisible(),
+      ]);
+    });
+  });
+
+  test('Wait for valid introspection status', async ({ page }) => {
```

Comment:
So this `test` continues in the same browser as previous `test`? Or it kills the browser and starts a new session?